### PR TITLE
OCPBUGSM-46219: OCPBUGSM-46220: Update base image golang version to 1.18.1

### PR DESCRIPTION
**Pull request by eliorerz**
Due to several CVE bugs we need to update golang version to >= v1.18.1.
https://bugzilla.redhat.com/show_bug.cgi?id=2077688
https://bugzilla.redhat.com/show_bug.cgi?id=2077689

This PR updates golang version to v1.18.1 and golangci to v1.46.0.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

/cc @eliorerz

### CODE REVIEW

File: internal/operators/lvm/lvm_operator.go [hunk 1]
```diff
@@ -10,8 +10,6 @@ import (
 	"github.com/openshift/assisted-service/internal/oc"
 	"github.com/openshift/assisted-service/internal/operators/api"
 	"github.com/openshift/assisted-service/models"
-
-	// "github.com/openshift/assisted-service/pkg/conversions"
```
Comment (nmagnezi) [inline internal/operators/lvm/lvm_operator.go:14]: remove this?

Comment (eliorerz) [reply internal/operators/lvm/lvm_operator.go:14]: Do we need that?

File: cmd/main.go [hunk 2]
```diff
@@ -704,19 +704,13 @@ func createS3Bucket(objectHandler s3wrapper.API, log logrus.FieldLogger) {
 
 func createStorageClient(deployTarget string, storage string, s3cfg *s3wrapper.Config, fsWorkDir string,
 	log logrus.FieldLogger, metricsAPI metrics.API, fsThreshold int) s3wrapper.API {
-	var storageClient s3wrapper.API
+	var storageClient s3wrapper.API = nil
 	if storage != "" {
 		switch storage {
 		case storage_s3:
 			storageClient = s3wrapper.NewS3Client(s3cfg, log)
-			if storageClient == nil {
```
Comment (nmagnezi) [inline cmd/main.go:712]: can you explain these changes?

File: internal/ignition/ignition.go [hunk 3]
```diff
@@ -1296,7 +1296,7 @@ func firstN(s string, n int) string {
 
 func (g *installerGenerator) runCreateCommand(ctx context.Context, installerPath, command string, envVars []string) error {
 	log := logutil.FromContext(ctx, g.log)
-	cmd := exec.Command(installerPath, "create", command, "--dir", g.workDir)
+	cmd := exec.Command(installerPath, "create", command, "--dir", g.workDir) //nolint:gosec
```
[the remark this answers is not in the document]
Comment (eliorerz) [reply internal/ignition/ignition.go:1299]: Updating `golangci-lint` cussed the following error
```
G204: Subprocess launched with a potential tainted input or cmd arguments (gosec)
```

File: cmd/main.go (hunk 2 above)
Comment (eliorerz) [reply cmd/main.go:712]: Sure.
`golangci-lint` noted that those `if` statements are redundant, but now when looking on those statements again, only `s3wrapper.NewFSClient` function will never be nil.
```
cmd/main.go:711:58: SA4023: this comparison is never true (staticcheck)
                        if storageClient = s3wrapper.NewS3Client(s3cfg, log); storageClient == nil {
                                                                              ^
cmd/main.go:715:87: SA4023: this comparison is never true (staticcheck)
                        if storageClient = s3wrapper.NewFSClient(fsWorkDir, log, metricsAPI, fsThreshold); storageClient == nil {
                                                                                                           ^
cmd/main.go:725:58: SA4023: this comparison is never true (staticcheck)
                        if storageClient = s3wrapper.NewS3Client(s3cfg, log); storageClient == nil {
```

Comment (eliorerz) [reply cmd/main.go:712]: https://github.com/openshift/assisted-service/blob/master/pkg/s3wrapper/filesystem.go#L32

Comment (eliorerz) [reply cmd/main.go:712]: I'll push fix for that

**Review by nmagnezi — APPROVED** [review]
/lgtm

**Comment by eliorerz** [comment]
Overriding broken jobs:

/override ci/prow/edge-e2e-metal-assisted-odf 
/override ci/prow/edge-e2e-metal-assisted-cnv
/override ci/prow/edge-e2e-metal-assisted-capi

**Pull request merged**